### PR TITLE
[Reviewer: Richard] Increase Sprout's timeout to Homestead to 750ms

### PR DESCRIPTION
**Pull request by sebrexmetaswitch**
This change is to accompany https://github.com/Metaswitch/homestead/pull/475

We expect that initial registers will take over 500ms in a GR deployment, so we need to increase the timeout Sprout specifies on its requests to Homestead. It defaulted to 500ms before, but this increases it to 750ms.

(All the other options in the constructor are the defaults from https://github.com/Metaswitch/cpp-common/blob/master/include/httpconnection.h#L31)

`make full_test` and the live tests pass.

### CODE REVIEW

**Comment by rkd-msw** [comment]
@pdavies - this might be interesting from a 4-site GR point of view (i.e. 100ms intra-site latency may break again then).

**Comment by richardwhiuk** [comment]
@rkd I think for 4 site GR we'll need to exploit some more parallelism in both Sprout and Homestead to avoid unacceptable latency.

**Comment by sebrexmetaswitch** [comment]
I've now made the timeout configurable via a new config option: `sprout_homestead_timeout_ms`. If not set, it will default to `diameter_timeout_ms + 550`. If that's not set, it will default to 750

I've live tested that changing these config options work (i.e. sprout picks them up correctly and passes them to the HssConnection)

File: sprout-base.root/etc/init.d/sprout [hunk 1]
```diff
@@ -102,6 +102,21 @@ get_settings()
...
+        #  - else:
+        #    - if we have diameter_timeout_ms set, use 550 + that
+        #    - else use 750
+        if [ -z "$sprout_homestead_timeout_ms" ];
+        then
+          if [ -z "$diameter_timeout_ms" ];
+          then
+            sprout_homestead_timeout_ms=750
+          else
+            sprout_homestead_timeout_ms=$((diameter_timeout_ms + 550))
```
Comment (richardwhiuk) [inline sprout-base.root/etc/init.d/sprout]: As discussed, we think this should be twice the diameter timeout.

File: src/main.cpp [hunk 2]
```diff
@@ -442,6 +444,7 @@ static void usage(void)
        "                            then the iFC is skipped over.\n"
        "     --http-acr-logging     Whether to include the bodies of ACR HTTP requests when they are logged \n"
        "                            to SAS\n"
+       "     --homestead-timeout    The timeout to use on HTTP requests to Homestead\n"
```
Comment (richardwhiuk) [inline src/main.cpp]: Should we say 'in ms' here?

**Pull request merged**